workflows/differentialabundance.nf

```diff
@@ -454,6 +482,32 @@ workflow DIFFERENTIALABUNDANCE {
             .mix(GSEA_GSEA.out.versions)
     }
 
+    if (params.gprofiler2_run) {
+
+        // For gprofiler2, use only features that are considered differential
+        ch_filtered_diff = FILTER_DIFFTABLE.out.filtered
+
+        if (!params.gprofiler2_background_file) {
+            // If deactivated, use empty list as "background"
+            ch_background = []
+        } else if (params.gprofiler2_background_file == "auto") {
+            // If auto, use input matrix as background
+            ch_background = CUSTOM_MATRIXFILTER.out.filtered.map{it.tail()}.first()
+        } else {
+            ch_background = Channel.from(file(params.gprofiler2_background_file, checkIfExists: true))
+        }
+
+        ch_gprofiler_gene_sets = ch_gene_sets
```

Comment:
I fixed this so as not to overwrite the global var. As mentioned above, this hack would not be necessary if we arranged it so that the module prioritises the organism and token options. We would just supply the gene sets channel(empty or otherwise) and allow the module to ignore them in favour of other options.

Reply:
Ah, good catch with the global!